tests/crypto/sha_tests.cpp

```diff
@@ -75,7 +75,7 @@ void test_big( const std::string& expected ) {
 
 template<typename H>
 void test_stream( ) {
-    H hash( TEST1 );
+    H hash = H::hash( TEST1 );
```

Comment:
This is different.

Reply:
Yes, it is. We believe there is a bug in sha_tests. You should look at the hash class constructor (ripemd160 for example) which accepts the serialized hash. TEST1 on the other hand is declared this way:
```
static const std::string TEST1("abc")
```
which is apparently not a ripemd160 hash. 

Anyways, the test itself is all about object serialization/deserialization and it should not affect its success if the hash object is correctly constructed indeed or is in some undefined state.